presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -946,6 +951,21 @@ protected Type visitExists(ExistsPredicate node, StackableAstVisitorContext<Void
             return BOOLEAN;
         }
 
+        @Override
+        protected Type visitQuantifiedComparisonExpression(QuantifiedComparisonExpression node, StackableAstVisitorContext<Void> context)
```

Comment:
Added a check if a type is orderable/comparable. Also, added cases to a unit test for it.
Note, that currently the change only supports orderable types for "=ALL" and "<>ANY". This check is made in SubqueryPlanner as it's not a general limitation, but is implementation specific.